bin/i18n/resources/pegasus/hourofcode.rb

```diff
@@ -28,3 +24,5 @@ def self.sync_out
     end
   end
 end
+
+Dir[File.expand_path('../hourofcode/**/*.rb', __FILE__)].sort.each {|file| require file}
```

Comment:
Interesting, could you explain why this `require` is moved to the end of the script? I have not seen this pattern yet.

Reply:
Since we require `bin/i18n/resources/pegasus/hourofcode.rb` and use constants from the module `HourOfCode` in 
`bin/i18n/resources/pegasus/hourofcode/sync_in.rb` and `bin/i18n/resources/pegasus/hourofcode/sync_out.rb`, classes `HourOfCode:SyncIn` and `HourOfCode::SyncOut` must be defined after the module `HourOfCode` to see it.

Reply:
Since we require `/pegasus/hourofcode.rb` and use constants from the module `HourOfCode` in
`/pegasus/hourofcode/sync_in.rb` and `/pegasus/hourofcode/sync_out.rb`,
classes `HourOfCode:SyncIn` and `HourOfCode::SyncOut` must be defined after the module `HourOfCode` to see it.

Reply:
Do we need to do the same in the rest of ParentModules (being `HourOfCode`, `Markdown`, `Mobile`, etc parent modules)?

Reply:
We need this only when we try to define a constant based on a parent constant.
We must do it for `HourOfCode`, I did this for the rest of Pegasus resources just for consistency and this.
https://github.com/code-dot-org/code-dot-org/blob/10b25df9119d4b57b2f21d163896a1bf6fc58ec3/bin/i18n/resources/pegasus/hourofcode/sync_in.rb#L13-L14